--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,7 +42,9 @@ target_include_directories(dublin-traceroute PUBLIC ${JSONCPP_INCLUDE_DIRS})
 
 #set_property(TARGET dublintraceroute PROPERTY CXX_STANDARD 11)
 #set_property(TARGET dublintraceroute PROPERTY CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 4.8.1)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+endif()
 
 if (NOT CMAKE_INSTALL_BINDIR)
     set(CMAKE_INSTALL_BINDIR "bin")
